### PR TITLE
Add check for Google Project Number validity 

### PR DESCRIPTION
**Pull request by mkapostolov**
The check is needed as the project number is sometimes mistaken with Google API Key.

### CODE REVIEW

File: everlive-push-hybrid/scripts/common.js [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
     };
-    
-    app.getYear = function () {
+
+    app.isInSimulatorCheck = function() {
+        return window.navigator.simulator === true;
+    };
+
+    app.isInCompanionAppCheck = function() {
+        if (typeof(isInCompanionApp) !== 'undefined') {
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/common.js]: Please leave the check only to

```
if(isInCompanionApp) {
 return true; 
} else {
return false;
}
```

File: everlive-push-hybrid/scripts/config.js [hunk 2]
```diff
@@ -17,17 +17,29 @@
...
+            '<p>Please go to <span>/scripts/config.js</span> and replace <strong>$ANDROID_PROJECT_NUMBER$</strong> whit the ' +
+            '<span>Google API project number</span> at the beginning of the file.</p>',
 
-        SUCCESS_TEXT: 'SUCCESS!<br /><br />The device has been registered for push notifications.<br /><br />',
+        WRONG_GOOGLE_API_PROJECT_NUMBER: '<h3>Wrong Google API Project Number!</h3> ' +
+            '<p>It appears that you have filled in wrong Google API project number. ' +
+            'The required project number consist only from digits, 12 in length.</p> ' +
+            '<p>Please go to <span>/scripts/config.js</span> and place a valid project number.</p>',
+
+        IN_SIMULATOR_TEXT: '<br /><br />RUNNING IN SIMULATOR!<br />Push notifications are NOT SUPPORTED when running the app on device simulator.',
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/config.js]: "You can only test the device registration workflow"

File: everlive-push-hybrid/scripts/config.js [hunk 3]
```diff
@@ -17,17 +17,29 @@
...
+
+        IN_COMPANION_APP_TEXT: '<br /><br />RUNNING IN COMPANION APP!<br /><br />' +
+            'Push notifications are NOT SUPPORTED when running the app in Telerik Platform Companion app.<br /><br /> ' +
+            'Please build the app as a stand-alone app package!',
+
+
+        SUCCESS_TEXT: 'SUCCESS!<br />The device has been registered for push notifications.<br /><br />',
         UNREGISTERED_TEXT: 'Device successfully unregistered.',
         EMULATOR_MODE: false
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/config.js]: Is this used somewhere

File: everlive-push-hybrid/scripts/app.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+
     app.everlive = new Everlive({
         appId: app.config.everlive.appId,
         scheme: app.config.everlive.scheme
     });
-        
-    var fixViewResize = function () {
-        
+
+    var fixViewResize = function() {
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/app.js]: Not needed anymore

File: everlive-push-hybrid/scripts/app.js [hunk 5]
```diff
@@ -20,21 +20,35 @@
...
+
+        if (app.isInSimulatorCheck()) {
+            $('#warning').html(app.constants.IN_SIMULATOR_TEXT);
+        }
+
         fixViewResize();
 
         var os = kendo.support.mobileOS,
-        	statusBarStyle = os.ios && os.flatVersion >= 700 ? 'black-translucent' : 'black';
+            statusBarStyle = os.ios && os.flatVersion >= 700 ? 'black-translucent' : 'black';
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/app.js]: Not needed anymore

File: everlive-push-hybrid/scripts/app.js [hunk 6]
```diff
@@ -20,21 +20,35 @@
...
+                return;
+            }
         }
-        
+
+        if (app.isInSimulatorCheck()) {
+            $('#warning').html(app.constants.IN_SIMULATOR_TEXT);
+        }
+
         fixViewResize();
```
Comment (AntonDobrev) [inline everlive-push-hybrid/scripts/app.js]: Not needed anymore

File: everlive-push-hybrid/scripts/common.js (hunk 1 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/common.js]: The proposed check does not work so will stay for now:
if (typeof(isInCompanionApp) !== 'undefined') {

File: everlive-push-hybrid/scripts/config.js (hunk 2 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/config.js]: Done

File: everlive-push-hybrid/scripts/config.js (hunk 3 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/config.js]: Not used. Will be removed.

File: everlive-push-hybrid/scripts/app.js (hunk 4 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/app.js]: Removed

File: everlive-push-hybrid/scripts/app.js (hunk 5 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/app.js]: Removed

File: everlive-push-hybrid/scripts/app.js (hunk 6 above)
Comment (mkapostolov) [inline everlive-push-hybrid/scripts/app.js]: Removed

**Comment by AntonDobrev** [comment]
:+1:

**Pull request closed without merging**